cob/cob.c

```diff
@@ -1444,62 +1433,69 @@ M0_INTERNAL void m0_cob_iterator_fini(struct m0_cob_iterator *it)
 	m0_free(it->ci_key);
 }
 
+static int cob_iterator_callback(struct m0_btree_cb *cb,
+				 struct m0_btree_rec *rec)
+{
+	struct m0_cob_iterator *it = cb->c_datum;
+	struct m0_cob_nskey    *nskey;
+
+	nskey = (struct m0_cob_nskey *)rec->r_key.k_data.ov_buf[0];
+
+	M0_ASSERT(m0_fid_eq(&nskey->cnk_pfid, m0_cob_fid(it->ci_cob)));
```

Comment:
The reason to do ASSERT and not returning -ENOENT is if we hit assert at this stage, then we might need to do some changes in code. Because, in old code where we are using the cursor, we are moving the cursor to next and then checking for fid. So, if fid is not equal we are not copying data to it->ci_key, but moving cursor. As this functionality can not be supported with the current implementation, if we assert , we should fix this implementation.

Reply:
This will also work, I will make this changes.